mempool/v1/mempool.go

```diff
@@ -233,7 +235,7 @@ func (txmp *TxMempool) CheckTx(tx types.Tx, cb func(*abci.Response), txInfo memp
 	// Invoke an ABCI CheckTx for this transaction.
 	rsp, err := txmp.proxyAppConn.CheckTxSync(abci.RequestCheckTx{Tx: tx})
 	if err != nil {
-		txmp.cache.Remove(tx)
+		txmp.cache.Remove(tx.Key())
```

Comment:
Same as above. There is already a `txKey` you can use

Reply:
txKey is computed in the anonymous function above